pkg/data/management/role_data.go

```diff
@@ -92,6 +92,7 @@ func addRoles(wrangler *wrangler.Context, management *config.ManagementContext)
 	// restricted-admin will get cluster admin access to all downstream clusters but limited access to the local cluster
 	restrictedAdminRole := addUserRules(rb.addRole("Restricted Admin", "restricted-admin"))
 	restrictedAdminRole.
+		addRule().apiGroups("").resources("secret").verbs("create").
```

Comment:
More of a question really, but should this not be within the `if settings.RestrictedDefaultAdmin.Get() == "true"` in `L112`?

Your change seems like the safer approach from a backwards compatibility. But I am not sure why we add all other rules to this user when restricted admin is "disabled".

Reply:
I believe that specific check is if the bootstrap admin should be given the `restricted-admin` role. Normally it would give them the full admin role. I'm guessing it's used for hosted rancher?
We haven't yet removed `restricted-admin` from rancher, which is why we still always create the global-role, even if we don't want people using it. It's only when that setting is on that we give the role slightly more privileges.